src/redux/rockets/rocketSlice.js

```diff
@@ -0,0 +1,65 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+export const getRockets = createAsyncThunk('rockets/getRockets', async () => {
+  try {
+    const responce = await axios.get('https://api.spacexdata.com/v4/rockets');
+    const data = responce.data.map((rocket) => ({
+      id: rocket.id,
+      image: rocket.flickr_images[0],
+      name: rocket.name,
+      description: rocket.description,
+      reserved: false,
+    }));
+    return data;
+  } catch (error) {
+    return error;
+  }
+});
+
+export const rocketSlice = createSlice({
+  name: 'rockets',
+  initialState: {
+    rockets: [],
+    isLoading: false,
+    hasError: false,
+  },
+  reducers: {
+    reserveRocket: (state, action) => {
+      state.rockets = state.rockets.map((rocket) => {
+        if (rocket.id !== action.payload) {
+          return rocket;
+        }
+        return { ...rocket, reserved: true };
+      });
+    },
+    cancelRocket: (state, action) => {
+      state.rockets = state.rockets.map((rocket) => {
+        if (rocket.id !== action.payload) {
+          return rocket;
+        }
+        return { ...rocket, reserved: false };
+      });
+    },
+  },
```

Comment:
- [ ] Please 🙏🏽 add tests for your functions reserve and cancel, it is a project requirement for this capstone 👇🏽 

> NOTE: Make sure to add tests for all your components and functions.

Reply:
Please any suggestions on how I can do the test.

Reply:
![Screenshot 2023-08-31 005535](https://github.com/SnakyMz/space-hub/assets/115364362/73eba9b0-dc5a-42e6-969f-f715ee591cae)
Changes as suggested